src/mpc-hc/MainFrm.cpp

```diff
@@ -1871,6 +1879,11 @@ void CMainFrame::OnTimer(UINT_PTR nIDEvent)
                     m_wndSubresyncBar.SetTime(rtNow);
                     m_wndSubresyncBar.SetFPS(m_pCAP->GetFPS());
                 }
+
+                const CAppSettings& s = AfxGetAppSettings();
+                if (s.bAutoUploadSubtitles && (rtNow / rtDur == 90) && !m_pSubStreams.IsEmpty() && s.fEnableSubtitles && m_pCAP && m_pCAP->GetSubtitleDelay() == 0) {

```

Comment:
About the exact timer magic... std::vector<bool> with like 30sec granuality. Make true if pos hits the position. Then analyze. Probably with int accumulator that counts false->true transitions.